src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6844,6 +6844,17 @@ final int peekTokenType() throws SQLServerException
 		return currentPacket.payload[payloadOffset] & 0xFF;
 	}
 
+
+	final short peekStatusFlag() throws SQLServerException {
+		// skip the current packet(i.e, TDS packet type) and peek into the status flag (USHORT)
+		if (payloadOffset + 3 <= currentPacket.payloadLength) {
```

Comment:
ensurePayload() makes sense only for packet type. Since we already have determined the packet type to be TDS_DONE, there must be a status flag that follows as per TDS standards.